### DISCUSSION TITLE
Innermost Surround Feature

### DISCUSSION THREAD

**Discussion by ariel-frischer** [kylechui/nvim-surround#237] [Ideas]
Hello,
Commonly I'm wanting to either delete, yank, or change the innermost pair at the cursors position. I want to make life easier. instead of specifying which pair like `ys(` or `ys{`, Id like the plugin to find the pair innermost to the cursor, and use that. So lets give it some letter `z`, `ysz` would yank inside parenthesis if cursor is inside parenthesis first, otherwise brackets if cursor is inside brackets first, otherwise quotes, etc... Would be nice to extend this for deletion and changing as well.

**Comment by kylechui** [comment]
You already can change/delete the nearest surround via aliases, i.e. `dss` or `css[char]`. Yanking would require the generation of new motions, which at the moment is not within the scope of the plugin.

You can try the plugin that XXiaoA made; it's linked at the bottom of #135.

**Comment by ariel-frischer** [comment]
Oof, thank you! It works! My bad for not reading thoroughly enough. This makes things much easier. 🙏🏽
